Use emacs-module from /usr/include

--- a/elisp.h
+++ b/elisp.h
@@ -1,7 +1,7 @@
 #ifndef ELISP_H
 #define ELISP_H
 
-#include "emacs-module.h"
+#include <emacs-28-vcs/emacs-module.h>
 #include "vterm.h"
 
 // Emacs symbols


--- a/vterm-module.h
+++ b/vterm-module.h
@@ -1,7 +1,7 @@
 #ifndef VTERM_MODULE_H
 #define VTERM_MODULE_H
 
-#include "emacs-module.h"
+#include <emacs-28-vcs/emacs-module.h>
 #include <inttypes.h>
 #include <stdbool.h>
 #include <vterm.h>

